### PR TITLE
integration/spanner: expand the add/drop integration test

### PR DESCRIPTION
**Pull request by tmc**
This brings this test closer in line with the upstream test.

### CODE REVIEW

File: internal/integration/spanner_test.go [hunk 1]
```diff
@@ -30,7 +30,7 @@ type spannerTest struct {
 }
 
 var spannerTests = map[string]*spannerTest{
-	"spanner-emulator": {port: 9020},
+	"spanner-emulator": {port: 9010},
```
Comment (tmc) [inline internal/integration/spanner_test.go:33]: unrelated but this reflects the gRPC port that is used for the spanner emulator traffic.

File: internal/integration/spanner_test.go [hunk 2]
```diff
@@ -221,7 +240,7 @@ func (t *spannerTest) migrate(changes ...schema.Change) {
 func (t *spannerTest) dropIndexes(names ...string) {
 	t.Cleanup(func() {
 		for _, idx := range names {
-			_, err := t.db.Exec("DROP INDEX " + idx)
+			_, err := t.db.ExecContext(context.Background(), "DROP INDEX "+idx)
```
Comment (tmc) [inline internal/integration/spanner_test.go:243]: this is added to tighten conformance to the `schema.ExecQuerier` interface.

File: internal/integration/spanner_test.go [hunk 3]
```diff
@@ -232,12 +251,28 @@ func (t *spannerTest) dropIndexes(names ...string) {
 	})
 }
 
+// dropConstraints drops foreign keys in the forms of refs. A ref is a period-separated "${table}.${constraint}"
+func (t *spannerTest) dropConstraints(refs ...string) {
+	t.Cleanup(func() {
+		for _, ref := range refs {
+			parts := strings.Split(ref, ".")
+			table, cstraint := parts[0], parts[1]
+			query := fmt.Sprintf("ALTER TABLE `%v` DROP CONSTRAINT `%v`", table, cstraint)
```
Comment (a8m) [inline internal/integration/spanner_test.go:260]:
```suggestion
			query := fmt.Sprintf("ALTER TABLE `%s` DROP CONSTRAINT `%s`", table, cstraint)
```

**Review by a8m — APPROVED** [review]

**Comment by tmc** [comment]
@a8m sorry I had just rebased this onto #1220, just  added your suggestion.

**Pull request merged**